enhancements/node-tuning/pao-in-nto.md

```diff
@@ -0,0 +1,272 @@
+
+## Motivation
+
+The Telco team maintains its own build and release pipelines and releases via a [Telco specific errata](https://red.ht/3boqjeS) that however targets the OpenShift product. So from a customer perspective PAO is part of OCP.
+
+- More and more customers are using PAO to configure their clusters for low latency or are being told to do so. The Telco team has so far managed to release PAO on the same day as OCP to maintain the illusion it is part of the product, but it costs a LOT of time to maintain the downstream build process.
+
+- The OCP [gcp-rt](https://prow.ci.openshift.org/?type=periodic&job=*gcp-rt*) lane tests OCP deployment on top of GCP with the real-time kernel enabled, but it only works [thanks to some pieces of PAO tuning](https://github.com/openshift/release/pull/18870). However the tuning is a pre-rendered snippet, not linked to the operator, because the deployment does not know about PAO.
+
+- PAO is closely tied to some OCP components
+([Cluster Node Tuning Operator](https://github.com/openshift/cluster-node-tuning-operator))
+and depends on the OCP version. So we release PAO for every OCP minor version (since 4.4) and together with some of the OCP Z-streams.
+The upgrade procedure is tricky though, first OCP needs to be upgraded and then PAO via an OLM channel change.
+This is needed to make sure the necessary functionality is present in the running NTO.
+
+- The new [Workload partitioning](https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md)
+feature seeks to use PAO as the configuration tool.
+However Workload partitioning needs to be enabled at install phase and PAO CRDs are not yet available at that time.
+[A "hacky" procedure](https://github.com/openshift/enhancements/pull/753) is needed to overcome this.
+
+### Goals
+
+We (the telco team) would like to make PAO part of OpenShift. There are multiple paths we can take to achieve this, I am going to describe the one we prefer: **PAO becoming a sub-controller in NTO.**
+
+The initial plan is to keep all logic and custom resources intact, we want to start with simply moving the code over.
+
+### Non-Goals
+
+Logic changes or tighter integration with other MCO controllers are not expected to be part of this initial move. Just the delivery mechanism wrt OCP.
+
+## Proposal
+
+To be really clear here, the Telco team will handle all the necessary changes to both codebases, however we will need support in the form of PR reviews and design consultations.
+
+### Code
+
+PAO code would be moved into the NTO repository as a separate controller. The Telco team would keep ownership as it has the necessary domain knowledge needed to implement the low latency tuning.
+
+The code currently uses Operator SDK libraries in some of the tests and this might need to be changed.
```

Comment:
Many of the core operators use either `client-go` directly or moved to https://github.com/openshift/library-go .  You may want to investigate and look into that.  My long-term plan for NTO is to evaluate a transition from directly using `client-go` to using https://github.com/openshift/library-go instead.